lib/matplotlib/tests/test_determinism.py

```diff
@@ -8,13 +8,46 @@
 import pytest
 
 import matplotlib as mpl
-import matplotlib.testing.compare
 from matplotlib import pyplot as plt
-from matplotlib.testing._markers import needs_ghostscript, needs_usetex
+from matplotlib.cbook import get_sample_data
+from matplotlib.collections import PathCollection
+from matplotlib.image import BboxImage
+from matplotlib.offsetbox import AnchoredOffsetbox, AuxTransformBox
+from matplotlib.patches import Circle, PathPatch
+from matplotlib.path import Path
 from matplotlib.testing import subprocess_run_for_testing
+from matplotlib.testing._markers import needs_ghostscript, needs_usetex
+import matplotlib.testing.compare
+from matplotlib.text import TextPath
+from matplotlib.transforms import IdentityTransform
 
 
-def _save_figure(objects='mhi', fmt="pdf", usetex=False):
+class PathClippedImagePatch(PathPatch):
```

Comment:
Why do we need a custom class here?

Reply:
(this is going to take me a while to get around to, but I'll confirm results when possible)

Reply:
oh, I did not realize this was pulled from an example, if that is the case can you move it to in the function where it is used and add a note "lifted from example"?

Reply:
Class definition relocated, and an explanatory comment added alongside.

I haven't yet attempted to simplify/refactor the code itself to remove the class entirely - it may be possible; figuring out whether the bbox redraw is required (and how to catch that event without implementing a custom pathpatch, if so) seems to be the main question there.